x-pack/plugins/remote_clusters/server/routes/api/get_route.ts

```diff
@@ -35,15 +36,32 @@ export const register = (deps: RouteDependencies): void => {
       const clustersByName = await clusterClient.asCurrentUser.cluster.remoteInfo();
       const clusterNames = (clustersByName && Object.keys(clustersByName)) || [];
 
-      // Retrieve the cluster information for all the configured remote clusters.
-      // _none is never a valid index/alias/data-stream name so that way we can avoid
-      // using * which could be computationally expensive.
-      let clustersStatus: IndicesResolveClusterResponse = {};
-      if (clusterNames.length > 0) {
-        clustersStatus = await clusterClient.asCurrentUser.indices.resolveCluster({
-          name: clusterNames.map((cluster) => `${cluster}:_none`),
-        });
-      }
+      const clusterNamesChunks = chunk(clusterNames, CLUSTER_STATUS_CHUNK_SIZE);
+      const promises = clusterNamesChunks.map(async (clustersChunk) => {
```

Comment:
As best I understand, this is processing 10 requests at a time. While better than running all the requests at once, its possible a slow request would prevent the next batch of 10 requests from running. RxJS mergeMap offers a param that limits the number of concurrent requests.

Reply:
This is splitting the array of cluster names into chunks of 10 and creating one request for each chunk (that returns the staus of those 10 clusters) that is then called concurrently. So if there's 100 cluster names, 10 request will be fired concurrently.

If I understand correctly, what you are saying is that we could still split the `clusterNames` into chunks of 10 and processing them concurrently up to MAX_CONCURRENT_REQUESTS at a time. I'm not that well versed with rxjs but perhaps something like this?

```typescript
  const clusterNamesChunks = chunk(clusterNames, CLUSTER_STATUS_CHUNK_SIZE);

  return new Promise((resolve, reject) => {
    from(clusterNamesChunks)
      .pipe(
        mergeMap(async (clustersChunk) => {
          try {
            return await resolveClusterStatus(clustersChunk);
          } catch (error) {
            // Return null to handle the error for this chunk but continue with others
            return null;
          }
        }, MAX_CONCURRENT_REQUESTS),
        toArray()
      )
      .subscribe({
        next: (resolvedClusterStatus) => {
          // Flatten the results and filter out any null values
          const flattenedClusterStatus = resolvedClusterStatus.flat().filter(Boolean);
          // Combine the resolved cluster status into a single object
          const clusterStatus = assign({}, ...flattenedClusterStatus);

          // Resolve the Promise with the clusterStatus result
          resolve(clusterStatus);
        },
        error: (error) => {
          reject(error);
        }
      });
  });

```

Reply:
I misunderstood how the requests were bring chunked and was making this a bit more complex than it needs to be for the issue being addressed. It would be nice to address https://github.com/elastic/kibana/issues/194834 since we're looking at the same code but its best done in another PR.